torch/onnx/_internal/fx/fx_onnx_interpreter.py

```diff
@@ -294,40 +371,159 @@ def _wrap_fx_args_as_onnxscript_args(
+        elif node.op == "get_attr":
+            self.get_attr(
+                node,
+                onnxscript_graph,
+                fx_name_to_onnxscript_value,
+                fx_graph_module,
+            )
+        elif node.op == "call_function":
+            self.call_function(
+                node,
+                torchscript_tracer,
+                fx_name_to_onnxscript_value,
+                onnxfunction_dispatcher,
+                op_level_debug,
+            )
+        elif node.op == "call_method":
+            self.call_method(node)
+        elif node.op == "call_module":
+            self.call_module(node)
+        elif node.op == "output":
+            self.output(node, onnxscript_graph, fx_name_to_onnxscript_value)
+        else:
+            raise RuntimeError(f"Found node type not defined in torch.fx: {node.op}")
+
+    @diagnostics.diagnose_call(diagnostics.rules.atenlib_fx_to_onnx)
+    @_beartype.beartype
+    def run(
+        self,
+        fx_graph_module: torch.fx.GraphModule,
+        onnxfunction_dispatcher: onnxfunction_dispatcher.OnnxFunctionDispatcher,
+        op_level_debug: bool,
+    ) -> onnxscript_graph_building.TorchScriptGraph:
+        """Analyze all FX nodes and trigger their ONNX translation.
+
+        Args:
+            fx_graph_module: FX graph module to be translated.
+            onnxfunction_dispatcher: ONNX function dispatcher.
+            op_level_debug: Whether to enable op-level debug.
+        """
+        onnxscript_graph = onnxscript_graph_building.TorchScriptGraph()
+        torchscript_tracer = onnxscript_graph_building.TorchScriptTracingEvaluator(
+            onnxscript_graph
         )
-        location = _location_from_fx_stack_trace(node_stack_trace)
-        if location is not None:
-            diagnostic.with_location(location)
+        fx_name_to_onnxscript_value = {}
+
+        # TODO: Fix FakeTensorMode limitation asap
+        # We want to pass list of ints and floats to TorchScript graph correctly
+        # in _export_fx_to_ts, so we must disable FakeTensorMode. Otherwise, graph may
+        # receive FakeTensor and results runtime error. In addition, TorchScript-based
+        # ONNX exporter used in _ts_graph_to_onnx_model_in_protobuf is not compatible
+        # with FakeTensorMode.
+        with torch.utils._mode_utils.no_dispatch():
+            # node_fixed_shape is only used on op_level_debug purpose.
+            for node in fx_graph_module.graph.nodes:
+                self.run_node(
+                    node,
+                    fx_graph_module,
+                    onnxfunction_dispatcher,
+                    op_level_debug,
+                    onnxscript_graph,
+                    torchscript_tracer,
+                    fx_name_to_onnxscript_value,
+                )
+
+        return onnxscript_graph
 
-    if node.op == "placeholder":
+    @diagnostics.diagnose_call(
```

Comment:
This spawns two diagnostics for the single fx node, since the rule `fx_node_to_onnx` is already decorated at `run_node`.

For the sake of minimizing code changes in this PR, I'd recommend removing the decorator. 

If you think a finer grained diagnostic is better, then we can discuss if we should create new rules, e.g. `fx_node_placeholder_to_onnx`, `fx_node_call_function_to_onnx`. I don't think it is necessary. But regardless I do not recommend doing that in this PR.

Reply:
Will leave the one from `run_node` for now. 

Maybe there are improvements on the diagnostic API to prevent function wrapping for logging that we can discuss after this PR.